src/Google/Ads/GoogleAds/Lib/V14/GoogleAdsClientBuilder.php

```diff
@@ -415,6 +416,48 @@ public function validate()
                 'The gRPC channel credential can only be set when the transport is "grpc".'
             );
         }
+        // Check if the version of the grpc extension installed by Composer is greater than that of
+        // the extension installed by PECL, throw an exception to remind the user to upgrade.
+        $grpcPackageVersion = floatval(phpversion('grpc'));
```

Comment:
Using `floatval` here confused me a bit because I didn't know how it worked with versions with more than a dot in it, e.g. 1.1.1.
I looked at [the reference doc for it](https://www.php.net/manual/en/function.floatval.php) and found out it only considers the last dot, so version 1.1.1 would be parsed as 11.1.
Since this is the case, would you think using `intval` rather than `floatval` here and on line 457 would achieve the same behavior and generate less confusion in a reader?

(I know, this is a very picky nit 🤣 )

Reply:
`intval` wouldn't work because the version difference is mostly at the second or third digit.

From what I tested, 1.1.1 and 1.1.2 would be converted to 1.1, so it does no harm here.
But I think we can just convert them to string and let the string comparison do the work instead, which should be more accurate.

WDYT?

Reply:
Good catch. version_compare is probably the best here.